File: software/examples/csharp/ExampleButtonCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your LCD 20x4 Bricklet

	// Callback function for button pressed callback
	static void ButtonPressedCB(BrickletLCD20x4 sender, byte button)
	{
		Console.WriteLine("Button Pressed: " + button);
	}

	// Callback function for button released callback
	static void ButtonReleasedCB(BrickletLCD20x4 sender, byte button)
	{
		Console.WriteLine("Button Released: " + button);
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletLCD20x4 lcd = new BrickletLCD20x4(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register button pressed callback to function ButtonPressedCB
		lcd.ButtonPressed += ButtonPressedCB;

		// Register button released callback to function ButtonReleasedCB
		lcd.ButtonReleased += ButtonReleasedCB;

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/java/ExampleButtonCallback.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletLCD20x4;

public class ExampleButtonCallback {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;
	private static final String UID = "XYZ"; // Change to your UID

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletLCD20x4 lcd204 = new BrickletLCD20x4(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Add button pressed listener
		lcd204.addButtonPressedListener(new BrickletLCD20x4.ButtonPressedListener() {
			public void buttonPressed(short buttonPressed) {
				System.out.println("Button Pressed: " + buttonPressed);
			}
		});

		// Add button released listener
		lcd204.addButtonReleasedListener(new BrickletLCD20x4.ButtonReleasedListener() {
			public void buttonReleased(short buttonReleased) {
				System.out.println("Button Released: " + buttonReleased);
			}
		});

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
